test/Npgsql.Tests/SyncOrAsyncTestBase.cs

```diff
@@ -0,0 +1,24 @@
+using NUnit.Framework;
+
+namespace Npgsql.Tests
+{
+    [TestFixture(ExecutionMode.Synchronous)]
+    [TestFixture(ExecutionMode.Asynchronous)]
+    public abstract class SyncOrAsyncTestBase : TestBase
```

Comment:
Not sure it's a good idea to introduce a base class, since a test class can only have one superclass (and the value here is quite low)... But we can keep it this way for now and see later.